File: ProblemSet6/sentimental-mario-more/mario.py
from cs50 import get_int

while True:
    height = get_int("Height: ");
    if (1 <= height <= 8):
        break

for i in range(height):
    for _ in range(height - i - 1):
        print(" ", end = "")
    for _ in range(i + 1):
        print("#", end = "")
    for _ in range(2):
        print(" ", end = "")
    for _ in range(i + 1):
        print("#", end = "")
    print()

File: ProblemSet1/mario-less/mario.c
#include <stdio.h>
#include <cs50.h>

int main() {
    int length = 0;
    int temp = 0;
    do {
        length = get_int("Height: ");
    }while(length <= 0);

    for (int i = 0; i < length; i++) {
        // length每个循环都不会变，所以要创建一个中间变量让他每行减一
        temp = length - i;
        for (int k = temp - 1; k > 0; k--) {
            printf(" ");
        }

        for (int j = 0; j <= i; j++) {
            printf("#");
        }
        printf("\n");
    }

    return 0;
}